lerobot/common/datasets/utils.py

```diff
@@ -61,17 +61,35 @@ def unflatten_dict(d, sep="/"):
     return outdict
 
 
-def hf_transform_to_torch(items_dict: dict[torch.Tensor | None]):
+def hf_transform_to_torch(
+    items_dict: dict[torch.Tensor | None],
+    lang_tokenizer_name: str = "t5-small",
+):
     """Get a transform function that convert items from Hugging Face dataset (pyarrow)
     to torch tensors. Importantly, images are converted from PIL, which corresponds to
     a channel last representation (h w c) of uint8 type, to a torch image representation
     with channel first (c h w) of float32 type in range [0,1].
     """
+    # tokenize language instructions if it exists
+    if "language_instruction" in items_dict:
+        from transformers import AutoTokenizer
```

Comment:
Several remarks here:
- Is there any way we can avoid adding transformer as a dependency if it's just for a tokenizer, at least for now? Ok if we can't, just asking as it's quite heavy. It should at least be an optional (extra) dependency, and we should specify clearly which datasets need it so that the would user know whether he/she should install it or not.
- I don't see any sign of transformers being added to the projects' dependency, which is why it's failing on the CI. To add a dependency:
```diff
# pyproject.toml

[tool.poetry.dependencies]
+ transformers = {version = ">=4.44.1", optional = true}

[tool.poetry.extras]
+ language = ["transformers"]
```
Then run
```bash
poetry lock --no-update
```

This will make `transformers` installable with lerobot as an extra:
```bash
# using pip
pip install -e ".[language]"

# using poetry
poetry install . --extras "language"
```
- As it would be an optional dependency, it should not be imported without "safety measures", e.g.:
```python
try:
    importlib.import_module(transformers)
except ModuleNotFoundError as e:
    print(
        f"transformers is not installed. Please install it with `pip install 'lerobot[language]'`"
    )
    raise e
```
- For the CI to pass, you would need to use the same trick we use in e.g. [this test](https://github.com/huggingface/lerobot/blob/996468bcceafe6688a74d321550fced20f8ec83b/tests/test_push_dataset_to_hub.py#L265) where we use a `@require_package_arg` to specify which packages are required for the test to be run (and the test is skipped otherwise if it's not installed)

Don't hesitate to ping me or send me a DM if any of this isn't clear ;)

Reply:
Thanks @aliberts ! it is super clear. 

But I am considering just removing it. We can leave up to the user to choose the method of how to address language conditioned tasks. 

We can just ignore the 'str' instances in `utils.hf_transform_to_torch`. What do you think?